### PR TITLE
Custom locations

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @esoergel 

http://manage.dimagi.com/default.asp?238886
fixes location custom data
cc: @proteusvacuum

### CODE REVIEW

File: corehq/apps/locations/tests/test_location_import.py [hunk 1]
```diff
@@ -95,6 +95,37 @@ def test_import_with_invalid_location_id(self):
         with self.assertRaises(ResourceNotFound):
             Location.get('i-am-invalid')
 
+    def test_import_with_custom_data(self):
+        """
+        When importing with a invalid location id, import_location should not
+        create a new location
```
Comment (benrudolph) [inline corehq/apps/locations/tests/test_location_import.py]: need to change this...

File: corehq/apps/locations/bulk.py [hunk 2]
```diff
@@ -225,6 +225,9 @@ def no_changes_needed(self, existing, form_data, consumption):
         ):
             return False
 
+        if get_prefixed(form_data) != existing.metadata:
```
Comment (czue) [inline corehq/apps/locations/bulk.py]: does `JSONField` have normal equality checks with python dicts?

**Review by czue — APPROVED** [review]
nice tests. :snowman_with_snow:

File: corehq/apps/locations/tests/test_location_import.py [hunk 3]
```diff
@@ -95,6 +95,37 @@ def test_import_with_invalid_location_id(self):
...
+        # Ensure metadata change works
+        data = {
+            'name': 'pablo',
+            'site_code': 'colombia',
+            'data': {
+                'drug': 'cocaine',
+            },
+        }
+        result = import_location(self.domain.name, 'state', data)
+        result = import_location(self.domain.name, 'state', data)
```
Comment (czue) [inline corehq/apps/locations/tests/test_location_import.py]: did you mean to include this twice?

File: corehq/apps/locations/bulk.py (hunk 2 above)
Comment (benrudolph) [inline corehq/apps/locations/bulk.py]: ah good question, not sure

Comment (benrudolph) [inline corehq/apps/locations/bulk.py]: https://github.com/dimagi/commcare-hq/pull/13475/commits/ab5c5f4dea6d69c3a4d8bae5b29ca4edf9989a3c

File: corehq/apps/locations/tests/test_location_import.py (hunk 3 above)
Comment (benrudolph) [inline corehq/apps/locations/tests/test_location_import.py]: no, thanks 🐦

**Pull request merged**